packages/front-end/pages/datasources/queries/[did].tsx

```diff
@@ -0,0 +1,203 @@
+import React, { useState } from "react";
+import {
+  FaCheck,
+  FaCircle,
+  FaExclamationTriangle,
+  FaSquare,
+} from "react-icons/fa";
+import { useRouter } from "next/router";
+import { ago, datetime } from "shared/dates";
+import { QueryInterface } from "back-end/types/query";
+import { capitalize } from "lodash";
+import { useSearch } from "@/services/search";
+import Tooltip from "@/components/Tooltip/Tooltip";
+import useApi from "@/hooks/useApi";
+import PageHead from "@/components/Layout/PageHead";
+import LoadingOverlay from "@/components/LoadingOverlay";
+import { useDefinitions } from "@/services/DefinitionsContext";
+import Modal from "@/components/Modal";
+import ExpandableQuery from "@/components/Queries/ExpandableQuery";
+
+const DataSourceQueries = (): React.ReactElement => {
+  const [modalData, setModalData] = useState<QueryInterface | null>(null);
+  const router = useRouter();
+  const { did } = router.query as { did: string };
+  const { getDatasourceById, ready, error: datasourceError } = useDefinitions();
+  const d = getDatasourceById(did);
+
+  const { data, error: queriesError } = useApi<{
+    queries: QueryInterface[];
+  }>(`/datasource/${did}/queries`);
+
+  const queries = data?.queries;
+
+  const { items, SortableTH } = useSearch({
+    items: queries || [],
+    defaultSortField: "createdAt",
+    localStorageKey: "datasourceQueries",
+    searchFields: ["status", "startedAt", "finishedAt"],
+  });
+
+  if (datasourceError || queriesError) {
+    return (
+      <div className="container pagecontents">
+        <div className="alert alert-danger">
+          {datasourceError ?? queriesError?.message}
+        </div>
+      </div>
+    );
+  }
+
+  if (!ready || !data) {
+    return <LoadingOverlay />;
+  }
+  if (!d) {
+    return (
+      <div className="container pagecontents">
+        <div className="alert alert-danger">
+          Datasource <code>{did}</code> does not exist.
+        </div>
+      </div>
+    );
+  }
+
+  if (!queries?.length) {
+    return (
+      <div className="container pagecontents p-4">
+        <div className="d-flex">
+          <h1>Data Source Queries</h1>
+        </div>
+        <p>No queries have been run on this Data Source.</p>
+      </div>
+    );
+  }
+
+  return (
+    <div className="container pagecontents">
+      {modalData && (
+        <Modal
+          open={true}
```

Comment:
Nit, possible to do in shorthand
```suggestion
          open
```